packages/x-date-pickers-pro/src/internal/models/fields.ts

```diff
@@ -3,7 +3,16 @@ import { SlotComponentProps } from '@mui/base/utils';
 import TextField, { TextFieldProps } from '@mui/material/TextField';
 import Stack, { StackProps } from '@mui/material/Stack';
 import Typography, { TypographyProps } from '@mui/material/Typography';
-import { BaseFieldProps } from '@mui/x-date-pickers/internals';
+import { BaseFieldProps, FieldSection } from '@mui/x-date-pickers/internals';
+
+export interface RangeFieldSection extends FieldSection {
```

Comment:
You can probably drop `DateRangeFieldSection` in favor of this `RangeFieldSection`